react-stopwatch-component/my-app/src/Stopwatch.tsx

```diff
@@ -0,0 +1,123 @@
+import { FaPlay, FaPause } from 'react-icons/fa';
+import './Stopwatch.css';
+import { useState } from 'react';
+
+export default function Stopwatch() {
+  const [time, setTime] = useState(0);
+  const [running, setRunning] = useState(false);
+  const [intervalId, setIntervalId] = useState<NodeJS.Timer>();
+
+  function handleStartStopClick() {
+    setRunning(!running);
+    if (running) {
+      clearInterval(intervalId);
+      setIntervalId(undefined);
+    } else {
+      setIntervalId(
+        setInterval(() => {
+          setTime((prev) => prev + 1);
+        }, 500)
+      );
+    }
+  }
+
+  function handleResetClick() {
+    if (!running) {
+      setRunning(!running);
```

Comment:
Oh, you set running to true? But then you clear the interval, so it won't be running. Do you get some unusual behavior? It seems like you would.